### ISSUE TITLE
Voice Over highlight doesn't move with the ruler

### ISSUE THREAD

**Issue by zepumph** [phetsims/gravity-force-lab#194]
@terracoda @jessegreenberg and I found that the VO focus highlight often doesn't "keep up" with the scenery Node. We should make sure that the css transforms of the PDOM are constantly staying in sync with the scenery Node (assigning @jessegreenberg to investigate this).

We then noticed that grabbing the ruler doesn't always work in mobile VO (gesture). This may have to do with the fact that scenery isn't keeping the VO highlight in the right place, but it also could be related to https://github.com/phetsims/gravity-force-lab/issues/189. I will solve that over there, and then we should come back here are retest to see if the out of sync VO highlight ever effects our ability to grab the ruler.

Note: The BASE Balloon VO highlight disappears when grabbing the balloon, and doesn't reappear again until you release the balloon, in which the VO highlight is in the right place around the balloon.  The balloon can move more after being released; in this case the VO highlight stays invisible until the balloon is done moving, and then it appears in the right place on the balloon once done moving.

Marking on hold until #189 is solved. We don't want to assume that this is a separate issue.

**Comment by jessegreenberg** [comment]
No longer on hold because I can easily see the element bounds are not correct on chrome. 

![image](https://user-images.githubusercontent.com/6396244/68616681-85388980-0493-11ea-8171-610ae48b3289.png)

**Comment by jessegreenberg** [comment]
When moving the ruler wiht the keyboard the positioning update is correct. When moving the ruler with the mouse/pointer positioning is incorrect.

**Comment by jessegreenberg** [comment]
>  by a mouse because an up event is caught over something else

We worked on this part today and have a fix committed. The positioning is correct on grab/release now and so this should be mostly working for the user. But the positioning still isn't updating while dragging. We tried to see if this was happening in BASE and it is not.

**Comment by jessegreenberg** [comment]
The elements should update when the node transform changes, but I notice that we are never hitting the listener on the TransformTracker.

**Comment by jessegreenberg** [comment]
We discussed this today with help from @jonathanolson. We decided that we will add a setter to Accessibility.js to specify the source Node whose transforms should control updates to the PDOM siblings.

We brainstormed some potential names for this field
```
  - transformTarget
  - transformTargetNode
  - transformTargetListeningNode
  - transformListenerNode
  - pdomTransformNode
  - sourceNode
  - transformSourceNode
  - pdomTransformSourceNode
  - transformSource
  - transformListenee
  - visualTransformer
  - visualTransformNode
  - matchVisualNode
  - matchPDOMToVisualNode
  - observeVisualNodeTransform
  - observeTransformNode
```

We like `pdomTransformSourceNode` at this time. It is long but this is a pretty low level/complicated function so we are OK with that.

We also considered doing "sneaky" things like trying to extend the trail that we observe to a more leaf level node, or update the PDOM sibling positioning every animation frame. These sounded too complicated or performance intensive to investigate further.

**Comment by terracoda** [comment]
Today in 2.2.0-dev.18 I noticed that the VO highlight moves tot eh ruler sometimes when the pink focus highlight also shows up. Once the pink focus highlight shows up, the black VO highlight seems to jump into place around the ruler. This happens randomly as far as I can tell.

**Comment by jessegreenberg** [comment]
Thanks @terracoda! I think that is to be expected at this time, hopefully https://github.com/phetsims/gravity-force-lab/issues/194#issuecomment-555711695 will have this fixed.

**Comment by jessegreenberg** [comment]
`pdomTransformSourceNode` was added to Accessibility.js in the above commit. I also added QUnit tests for it.

**Comment by jessegreenberg** [comment]
Used in inverse-square-law-common in the above commit. I tried to do exactly what we discussed in https://github.com/phetsims/gravity-force-lab/issues/194#issuecomment-555711695 but @zepumph would you please review these changes?

**Comment by zepumph** [comment]
This looks really awesome! I'm excited for these changes, especially in GFL

- [ ] Since `pdomTransformSourceNode` is only ever `Node|null`, this:
      `const transformSourceNode = this.node.pdomTransformSourceNode ? this.node.pdomTransformSourceNode : this.node;`
     can be simplified to `const transformSourceNode = this.node.pdomTransformSourceNode || this.node;`
- [ ] At first glance I'm a bit confused as to the division of labor for this commit between `AccessibleInstance` and `AccessiblePeer`. Can you speak to this at all? For example. Why does one have setPDOMTransformSourceNode, but the other has `updateTransformTracker`? Note though that I do like the idiomatic way that `Accessibility.setPDOMTransformSourceNode` calls over to the same `AccessiblePeer` function, as this is the pattern used throughout the file.

**Comment by terracoda** [comment]
In [2.2.0-dev.20](https://phet-dev.colorado.edu/html/gravity-force-lab/2.2.0-dev.20/phet/gravity-force-lab_all_phet.html) the rVO highlight now floows teh position of the ruler and it is also displaying snug around the ruler now, too.

**Comment by terracoda** [comment]
@zepumph and @jessegreenberg, if no code review is need here, I think we can close.

**Comment by jessegreenberg** [comment]
> ...can be simplified to...

Thanks!
> I'm a bit confused as to the division of labor for this commit between AccessibleInstance and AccessiblePeer. Can you speak to this at all?

I struggle with the separation of these two in general. AccessibleInstance is the "model" and AccessiblePeer is the "view", and I think of TransformTracker and its Trail as "model" components to be observed by the AccessiblePeer. And so I think AccessibleInstance should be responsible for updating the TransformTracker while AccessiblePeer is responsible for updating listeners on the TransformTracker.

**Comment by jessegreenberg** [comment]
https://github.com/phetsims/gravity-force-lab/issues/194#issuecomment-558656977

Great, thanks @terracoda!

**Comment by zepumph** [comment]
Sounds good to me. Ready to close?
